test/cdk-lib/bedrock/agent-action-group.test.ts

```diff
@@ -0,0 +1,158 @@
+
+// mock lambda.Code.fromDockerBuild()
+jest.mock('aws-cdk-lib/aws-lambda', () => {
+  const actualLambda = jest.requireActual('aws-cdk-lib/aws-lambda');
+  return {
+    ...actualLambda,
+    Code: {
+      ...actualLambda.Code,
+      fromDockerBuild: jest.fn(() => actualLambda.Code.fromInline('mockCode')),
+      fromAsset: jest.fn(() => actualLambda.Code.fromInline('mockCode')),
+    },
+  };
+});
+
+describe('AgentActionGroup', () => {
+  let app: cdk.App;
+  let stack: cdk.Stack;
+  let agent: bedrock.Agent;
+  let actionGroupFunction: lambda.Function;
+  const actionGroupAPISpec = 'mockApiSpec';
+
+  beforeEach(() => {
+    app = new cdk.App();
+    stack = new cdk.Stack(app, 'test-stack', {
+      env: {
+        account: '123456789012',
+        region: 'us-east-1',
+      },
+    });
+
+    agent = new bedrock.Agent(stack, 'Agent', {
+      foundationModel: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_V2_1,
+      instruction: 'You provide support for developers working with CDK constructs.',
+      idleSessionTTL: cdk.Duration.minutes(30),
+    });
+
+    actionGroupFunction = new lambda.Function(stack, 'ActionGroupFunction', {
+      code: lambda.Code.fromAsset('test/path'),
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      handler: 'index.handler',
+    });
+
+    NagSuppressions.addResourceSuppressions(
+      actionGroupFunction,
+      [
+        {
+          id: 'AwsSolutions-IAM4',
+          reason: 'ActionGroup Lambda uses the AWSLambdaBasicExecutionRole AWS Managed Policy.',
+        },
+      ],
+      true,
+    );
+  });
+
+  test('actionGroupExecutor invocation policy is configured', () =>{
+    new bedrock.AgentActionGroup(stack, 'ActionGroup', {
+      agent,
+      description: 'Use these functions to get information about the books in the Project Gutenburg library.',
+      actionGroupExecutor: actionGroupFunction,
+      actionGroupState: 'ENABLED',
+      apiSchema: bedrock.ApiSchema.fromInline(actionGroupAPISpec),
+    });
+
+    const template = Template.fromStack(stack);
+    template.hasResourceProperties('AWS::Lambda::Permission', {
+      Action: 'lambda:InvokeFunction',
+      Principal: 'bedrock.amazonaws.com',
+      SourceAccount: '123456789012',
+    });
+  });
+
+  test('Fails when both description and parentActionGroupSignature are provided', () => {
+    expect(() => {
+      new bedrock.AgentActionGroup(stack, 'ActionGroup', {
+        agent,
+        description: 'Use these functions to get information about the books in the Project Gutenburg library.',
+        parentActionGroupSignature: 'AMAZON.UserInput',
+        actionGroupExecutor: actionGroupFunction,
+        actionGroupState: 'ENABLED',
+        apiSchema: bedrock.ApiSchema.fromInline(actionGroupAPISpec),
+      });
+    }).toThrowError('Cannot specify both description and parentActionSignature');
+  });
+
+  // test('Fails when both apiSchema.payload and apiSchema.s3 are provided', () => {
```

Comment:
Is this commented code block still needed?

Reply:
Thanks for catching that. After adding the ApiSchema class, I don't have a way to test that. I'm removing it.